src/index.c

```diff
@@ -699,10 +699,15 @@ static IndexCriteriaTester *II_GetCriteriaTester(void *ctx) {
     return NULL;
   }
 
-  IICriteriaTester *ict = rm_malloc(sizeof(*ict));
   for (size_t i = 0; i < array_len(ic->its); ++i) {
+    IndexCriteriaTester *sub = IITER_GET_CRITERIA_TESTER(ic->its[i]);
+    if (!sub) {
+      return NULL;
+    }
     ic->testers = array_append(ic->testers, IITER_GET_CRITERIA_TESTER(ic->its[i]));
```

Comment:
So when you reach here you have the testers array, and then you readd all the tests?